sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -101,6 +101,7 @@ case class Alias(child: Expression, name: String)
   extends NamedExpression with trees.UnaryNode[Expression] {
 
   override type EvaluatedType = Any
+  override lazy val resolved = childrenResolved && !child.isInstanceOf[Generator]
```

Comment:
`Alias(Generator)` is not like the normal expression, and it will be transformed into `Generate(Generator, alias)`.